extensions/standard-processors/processors/GetTCP.h

```diff
@@ -212,29 +207,29 @@ class GetTCP : public core::Processor, public state::response::MetricsNodeSource
    * @param sessionFactory process session factory that is used when creating
    * ProcessSession objects.
    */
-  virtual void onSchedule(const std::shared_ptr<core::ProcessContext> &processContext, const std::shared_ptr<core::ProcessSessionFactory> &sessionFactory);
+  void onSchedule(const std::shared_ptr<core::ProcessContext> &processContext, const std::shared_ptr<core::ProcessSessionFactory> &sessionFactory) override;
 
-  void onSchedule(core::ProcessContext *processContext, core::ProcessSessionFactory *sessionFactory) {
-    throw std::exception();
+  void onSchedule(core::ProcessContext *processContext, core::ProcessSessionFactory *sessionFactory) override {
+    throw std::logic_error{"GetTCP::onSchedule(ProcessContext*, ProcessSessionFactory*) is unimplemented"};
```

Comment:
I kind of get the reason behind this `onSchedule` overload, but shouldn't we overload the raw-pointer-taking `onSchedule` (unless the processor wants to store the context or factory) and leave the `onSchedule(shared_ptr<ProcessContext>, shared_ptr<ProcessSessionFactory>)` alone (defaulting to forwarding to the raw-pointer-taking `onSchedule` in `Processor`), then we wouldn't have to throw at all

Reply:
Ideally yes, but at the moment GetTCP creates flow files as data arrives meaning that it needs to be able to create a ProcessSession whenever needed and shared ownership of ProcessSessionFactory is the way it achieves this.

Reply:
I see, I just glanced over it and missed the part where it stores the factory

Reply:
I also missed it until the compilation errors popped up. :smile: